src/traits/impls.rs

```diff
@@ -46,6 +48,51 @@ impl DoughnutApi for () {
     }
 }
 
+impl Signing for () {
+    fn sign_ed25519(&self, public_key: &[u8], private_key: &[u8], payload: &[u8]) -> Result<(), VerifyError> {
+        sign_ed25519(&public_key, &private_key, &payload);
+    }
+
+    fn sign_sr25519(&self, public_key: &[u8], private_key: &[u8], payload: &[u8]) -> Result<(), VerifyError> {
+        sign_sr25519(&public_key, &private_key, &payload);
+    }
+}
+
+#[cfg(feature = "std")]
+impl<'a> Signing for DoughnutV0<'a> {
+    fn sign_ed25519(&self, public_key: &[u8], private_key: &[u8], payload: &[u8]) -> Result<(), VerifyError> {
+        sign_ed25519(&public_key, &private_key, &payload);
+    }
+
+    fn sign_sr25519(&self, public_key: &[u8], private_key: &[u8], payload: &[u8]) -> Result<(), VerifyError> {
+        sign_sr25519(&public_key, &private_key, &payload);
+    }
+}
+
+#[cfg(feature = "std")]
+impl Signing for ParityDoughnutV0 {
+    fn sign_ed25519(&self, public_key: &[u8], private_key: &[u8], payload: &[u8]) -> Result<(), VerifyError> {
+        sign_ed25519(&public_key, &private_key, &payload);
+    }
+
+    fn sign_sr25519(&self, public_key: &[u8], private_key: &[u8], payload: &[u8]) -> Result<(), VerifyError> {
+        sign_sr25519(&public_key, &private_key, &payload);
+    }
+}
+
+#[cfg(feature = "std")]
+#[allow(unreachable_patterns)]
+impl Signing for Doughnut {
```

Comment:
@holygits Shall we implement signing for all above Doughnuts, `(), DoughnutV0, ParityDoughnutV0 `?

Reply:
Can start with `ParityDoughnutV0`, an impl for `()` will only be useful for tests. It's not used at the moment so would leave it out